Using the Telerik Nuget alongside SSO account (#20)
telerik/fiddler-core-docs: getting-started/nuget-sso.md

---

> title: Using Telerik NuGet with SSO Account

Reviewer: Could you consider using a title that matches the style of the other titles in the Getting Started section? For example, **Telerik NuGet and SSO Accounts**.

![image](https://user-images.githubusercontent.com/60105689/214592401-577634a2-9fdd-4e71-aa94-5f52faf0c465.png)

Alternatively, you can update the existing titles so that they use the -ing forms:
- Downloading Product Files
- Using Telerik NuGet
- Using Telerik NuGet with SSO Account

> description: Using Telerik Nuget alongside an SSO account

Reviewer: Aim at descriptions that are around 150 symbols long (SEO recommendation).
```suggestion
description: Learn how to use API keys for authentication with the Telerik NuGet server when you have an SSO account and cannot use a username and password. 
```

Instead of entering a username and password through a Visual Studio prompt or a CLI command, the NuGet key must be provided through the NuGet.Config file. 

You have the option to create a NuGet.Config file on the application level (it will overwrite any global NuGet.Config files and configuration will be applicable only for the specific application) or to use a global NuGet.Config file (it will be valid for all applications unless explicitly overwritten by an application NuGet.Config file). [Learn more details about NuGet.Config configuration and location on each operating system here...](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior)

- Close all instances of Visual Studio.

- Navigate to the directory where the NuGet.Config file resides. [Learn where is the location of NuGet.Config on each operating system here...](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#config-file-locations-and-uses)

- Open the NuGet.Config file or create one if the file does not exist.

- Add the Telerik NuGet server in the package source section through a custom key (in the demo case, the key is named **MyTelerikFeed**)

> - Add the generated Telerik NuGet key through the **packageSourceCredentials** section by adding a **Username** key with the value **api-key** and a **ClearTextPassword** key with the value of the generated Telerik NuGet key.

Reviewer: As this step consists of two actions, you can add them as a nested list for better visibility.
```suggestion
1. Add the generated Telerik NuGet key in the `packageSourceCredentials` section:
   1. Add a `Username` key with the value `api-key`
   1. Add a `ClearTextPassword` key with the value of the generated Telerik NuGet key.
```

>     An example of adding the Telerik NuGet server with an API key.

Reviewer: As the previous list items explain the content of the code snippet, you can remove this introductory sentence.

    ```XML
    <?xml version="1.0" encoding="utf-8"?>
    <configuration>
    <packageSources>
        <clear/>
        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
        <add key="MyTelerikFeed" value="https://nuget.telerik.com/v3/index.json" protocolVersion="3"/>
    </packageSources>
    <packageSourceCredentials>
        <MyTelerikFeed>
            <add key="Username" value="api-key" />
            <add key="ClearTextPassword" value="%MY_API_KEY%" />
        </MyTelerikFeed>
    </packageSourceCredentials>
    <!-- more options follow here -->
    </configuration>

    ```

>     It is crucial that you use the **ClearTextPassword** option (and not **Password**). Note that while you can directly copy/paste the Telerik NuGet key, it is not a good practice in terms of security. Once again, you can learn how to protect your api key from [the Lance blog post](https://www.telerik.com/blogs/announcing-nuget-keys) that we mentioned earlier.

Reviewer: ```suggestion
    >Always use the `ClearTextPassword` option (not `Password`). Note that while you can directly copy/paste the Telerik NuGet key, it is not a good practice in terms of security. You can learn how to protect your API key from [Lance's blog post on NuGet keys](https://www.telerik.com/blogs/announcing-nuget-keys).
```
